Automattic/jetpack
# Added functionality to add fake plugin and theme versions that will be detected by Jetpack Scan.

## Proposed changes:

* Add more ways to fake threats that can be detected by Jetpack Scan, specifically:
  * Threat caught by fuzzy hash
  * Fake an installation of a vulnerable plugin
  * Fake an installation of a vulnerable theme
  * Fake the WordPress version to be one that is vulnerable

As discussed here: p1719323414750979-slack-C029WFNV69M - this PR does not add a mechanism to make the site be detected as a Google Web Risk threat, as that will have to work differently.

### Other information:


## Jetpack product discussion



https://github.com/Automattic/jetpack-scan-team/issues/1230

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






* Spin up local environment and start jurassic tube
* Add the various threats via the Jetpack Debug Scan helper
* Run a scan on against your site and verify the threats are detected as expected
* Remove the threats again via the Jetpack Debug Scan helper
* Run a scan again and verify the threats are no longer detected

A combination of any number of different threats at a time makes the testing even more accurate.
